quodlibet/order/__init__.py

```diff
@@ -109,33 +109,34 @@ def __init__(self):
 
     def next(self, playlist, iter):
         if iter is not None:
-            self._played.append(playlist.get_path(iter).get_indices()[0])
+            self._played.append(iter)
```

Comment:
Adding a type annotation in the constructor would let readers know it's now a different type of list element which might be nice

Reply:
I've been hesitant to use GObject classes in type hints, because they are not usable for the type checker. This code at the moment infers the type `List[Unknown]`. If I write `self._played: List[Gtk.TreeIter] = []` it infers the same. In fact I can write `self._played: List[Gtk.TreeeeIter] = []` and get no error.
I'll add it of course if you prefer. I guess it functions just as well as a comment would.

Reply:
>  I guess it functions just as well as a comment would.

Yes - this is the win IMO (also MyPy is improving and the stubs hopefully will do too)